creusot/tests/should_succeed/selection_sort_i32.rs

```diff
@@ -0,0 +1,135 @@
+#![feature(type_ascription)]
+extern crate creusot_contracts;
+
+use creusot_contracts::std::*;
+use creusot_contracts::*;
+
+pub struct Ghost<T>(*mut T)
+where
+    T: ?Sized;
+
+impl<T> Model for Ghost<T> {
+    type ModelTy = T;
+    #[logic]
+    #[trusted]
+    fn model(self) -> Self::ModelTy {
+        panic!()
+    }
+}
+
+impl<T> Ghost<T> {
+    #[trusted]
+    #[ensures(@result === *a)]
+    fn record(a: &T) -> Ghost<T> {
+        panic!()
+    }
+}
+
+#[predicate]
+fn sorted_range(s: Seq<i32>, l: Int, u: Int) -> bool {
+    pearlite! {
+        forall<i: Int, j : Int> l <= i && i < j && j < u ==> s[i] <= (s[j])
+    }
+}
+
+#[predicate]
+fn sorted(s: Seq<i32>) -> bool {
+    pearlite! {
+        sorted_range(s, 0, s.len())
+    }
+}
+
+#[predicate]
+fn smaller_than(s: Seq<i32>, e: i32, l: Int) -> bool {
+    pearlite! {
+        forall<j: Int> l < j && j < s.len() ==>
+            e <= s[j]
+    }
+}
+
+#[predicate]
+fn larger_than(s: Seq<i32>, e: i32, u: Int) -> bool {
+    pearlite! {
+        forall<i: Int> 0 <= i && i < u && i < s.len() ==>
+            e >= s[i]
+    }
+}
```

Comment:
I wonder if these couldn't be combined? It seems like they are actually stating something about the partition of the sequence around `e`?

Reply:
Perhaps. They are used to state that everything in the sorted part is smaller than or equal to everything in the larger part, and vice versa. Probably possible to merge, but I'm not sure that it will become more readable/prettier/faster.